Don't call onError if an error occurs while processing avatar

martintoften opened the pull request:
Signal is throwing `PushNetworkException` with the message `thread interrupted` when calling 
`SignalServiceMessageReceiver::retrieveAttachment` in `FileUtil::writeAttachmentToFileFromMessageReceiver`

This actually happens quite often. So the group wouldn't be updated

csdodd commented on app/src/main/java/com/toshi/model/local/Group.java:
```diff
@@ -74,35 +76,39 @@ public Group(final List<User> members) {
 
-        return Single.just(this);
+        return processAvatar(group, messageReceiver)
+                .toSingleDefault(this);
```
This is really odd abuse of Rx. Is it possible to rewrite this in a way that is clearer in its intentions?

martintoften replied:
Yep. You don't like the `toSingleDefault(this)`?

csdodd commented on app/src/main/java/com/toshi/model/local/Group.java:
```diff
@@ -74,35 +76,39 @@ public Group(final List<User> members) {
-
             return Single.fromCallable(() -> {
                 final SignalServiceAttachmentPointer attachment = group.getAvatar().get().asPointer();
                 return FileUtil.writeAttachmentToFileFromMessageReceiver(attachment, messageReceiver);
```
Why is this returning `null`?

martintoften replied:
Signal can throw PushNetworkException when calling
SignalServiceMessageReceiver::retrieveAttachment in FileUtil::writeAttachmentToFileFromMessageReceiver. If this happens it returns null, which we didn't handle

csdodd replied:
And if it returns `null`; we just never bother trying to get the file again? Why does it throw the exception?

martintoften replied:
We could of course retry. 

I get the message `thread interrupted`, but I'm not sure why it happens

martintoften commented:
Updated

I Didn't find a better way of solving `updateFromSignalGroup`. Not sure how I could return the updated group without using `toSingle`

Added a retry so if an error happens while processing avatars, it will retry 3 times

csdodd commented on app/src/main/java/com/toshi/model/local/Group.java:
```diff
@@ -94,6 +94,7 @@ private Completable processAvatar(final SignalServiceGroup group, final SignalSe
                 return FileUtil.writeAttachmentToFileFromMessageReceiver(attachment, messageReceiver);
             })
             .flatMap(this::compressImage)
+            .retry(3)
```
Just don't bother retrying. If you don't know why the error is occurring then you don't know if this is a fix. Remove this.
